cfme/middleware/provider.py

```diff
@@ -79,6 +77,27 @@ def _form_mapping(self, create=None, **kwargs):
                 'hostname_text': kwargs.get('hostname'),
                 'port_text': kwargs.get('port')}
 
+    def navigate(self, detail=True):
```

Comment:
This should be called load_details. But I have the feel that the load_details is already defined in the parent classes. Please check that out.